client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -183,25 +250,41 @@ public ReaderGroupConfig build() {
                endingStreamCuts = Collections.emptyMap();
            }
 
+           //fetch StreamCuts with the specified default scope.
+           final Map<String, StreamCut> startCutsWithDefaultScope = getCutsWithDefaultScope(startingStreamCuts);
```

Comment:
I think that what you are trying to achieve with this is that a stream cut has streams that point to a default scope, and the exact scope will depend on the config they are passed to. Consequently, if I pass a cut with default scope to a config with default scope `x1`, this method will use the `x1` with this cut. If I instead pass a cut with default scope to a config with default scope `x2`, this method will use the `x2` with this cut. This is prone to error because the streams are different and it doesn't make sense to do it.  A stream cut should here is be tied a particular scope/stream, and as such, the scope should be fixed.

Reply:
(answered as part of the first comment).